celery_exporter/celery_state.py

```diff
@@ -0,0 +1,169 @@
+from collections import defaultdict
+from dataclasses import dataclass
+from enum import Enum
+from typing import Dict, Optional, Tuple
+
+from cachetools import LRUCache
+from celery import states
+
+CELERY_MISSING_DATA = "undefined"
+# name, state, runtime, queue
+CollectOutcome = Tuple[Optional[str], Optional[str], Optional[float], Optional[str]]
+# name, queue, latency
+LatencyOutcome = Tuple[Optional[str], Optional[str], Optional[float]]
+
+
+def is_task_event(kind: str) -> bool:
+    return "task" in kind
+
+
+class TaskState(Enum):
+    PENDING = states.PENDING
+    RECEIVED = states.RECEIVED
+    STARTED = states.STARTED
+    FAILURE = states.FAILURE
+    RETRY = states.RETRY
+    SUCCESS = states.SUCCESS
+    REVOKED = states.REVOKED
+    REJECTED = states.REJECTED
+    UNDEFINED = "UNDEFINED"
+
+    @classmethod
+    def from_event(cls, event_kind: str) -> "TaskState":
+        MAPPING = {
+            "sent": TaskState.PENDING,
+            "received": TaskState.RECEIVED,
+            "started": TaskState.STARTED,
+            "failed": TaskState.FAILURE,
+            "retried": TaskState.RETRY,
+            "succeeded": TaskState.SUCCESS,
+            "revoked": TaskState.REVOKED,
+            "rejected": TaskState.REJECTED,
+        }
```

Comment:
mentioned in Slack: will this dict be recreated on each fn call?

Reply:
original
```
In [5]: %timeit TaskState.from_event(event)
727 ns ± 4.27 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

updated version
```
In [4]: %timeit TaskState.from_event(event)
194 ns ± 0.625 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```